src/zino/tasks/bgpstatemonitortask.py

```diff
@@ -0,0 +1,425 @@
+        )
+
+        cisco_bgp_info = await self._get_bgp_info(mib_name="CISCO-BGP4-MIB", variables=variables)
+
+        if not cisco_bgp_info:
+            return None
+
+        for oid, result in cisco_bgp_info.items():
+            result["cbgpPeer2RemoteAddr"] = oid
+
+        cisco_bgp_info = self._transform_variables_from_specific_to_general(bgp_info=cisco_bgp_info, bgp_style="cisco")
+
+        return cisco_bgp_info
+
+    async def _get_general_bgp_info(self) -> Union[SparseWalkResponse, None]:
+        variables = (
+            "bgpPeerState",
+            "bgpPeerAdminStatus",
+            "bgpPeerRemoteAddr",
+            "bgpPeerRemoteAs",
+            "bgpPeerFsmEstablishedTime",
+        )
+
+        general_bgp_info = await self._get_bgp_info(mib_name="BGP4-MIB", variables=variables)
+
+        if not general_bgp_info:
+            return None
+
+        general_bgp_info = self._transform_variables_from_specific_to_general(
+            bgp_info=general_bgp_info, bgp_style="general"
+        )
+
+        return general_bgp_info
+
+    async def _get_bgp_info(self, mib_name: str, variables: Iterable[str]) -> Union[SparseWalkResponse, None]:
+        snmp = SNMP(self.device)
+        try:
+            bgp_info = await snmp.sparsewalk(
+                *((mib_name, var) for var in variables),
+                max_repetitions=3,
+            )
+        except NoSuchNameError as e:
+            _logger.info(f"router {self.device.name} misses BGP variables ({e})")
+            return None
+
+        cleaned_bgp_info = dict()
+
+        for oid, entry in bgp_info.items():
+            if len(oid) == 1:
+                cleaned_bgp_info[oid[0].prettyPrint()] = entry
+            else:
+                cleaned_bgp_info[oid[1].prettyPrint()] = entry
+
+        return cleaned_bgp_info
+
+    def _transform_variables_from_specific_to_general(
+        self, bgp_info: SparseWalkResponse, bgp_style: BgpStyle
+    ) -> SparseWalkResponse:
+        juniper_translation = [
+            ("peer_state", "jnxBgpM2PeerState"),
+            ("peer_admin_status", "jnxBgpM2PeerStatus"),
+            ("peer_remote_address", "jnxBgpM2PeerRemoteAddr"),
+            ("peer_remote_as", "jnxBgpM2PeerRemoteAs"),
+            ("peer_fsm_established_time", "jnxBgpM2PeerFsmEstablishedTime"),
+        ]
+        cisco_translation = [
+            ("peer_state", "cbgpPeer2State"),
+            ("peer_admin_status", "cbgpPeer2AdminStatus"),
+            ("peer_remote_address", "cbgpPeer2RemoteAddr"),
+            ("peer_remote_as", "cbgpPeer2RemoteAs"),
+            ("peer_fsm_established_time", "cbgpPeer2FsmEstablishedTime"),
+        ]
+        general_translation = [
+            ("peer_state", "bgpPeerState"),
+            ("peer_admin_status", "bgpPeerAdminStatus"),
+            ("peer_remote_address", "bgpPeerRemoteAddr"),
+            ("peer_remote_as", "bgpPeerRemoteAs"),
+            ("peer_fsm_established_time", "bgpPeerFsmEstablishedTime"),
+        ]
+
```

Comment:
These maps look like constant values, so why re-define them on every call to this function?

Also, in combination with the following if-else statement, they could be made somewhat more compact...